helm/cilium/templates/default-policies/job.yaml

```diff
@@ -0,0 +1,63 @@
+{{- if .Values.defaultPolicies.enabled -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cilium-create-default-policies
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: cilium-create-default-policies
+  annotations:
+    helm.sh/hook: post-install
+    helm.sh/hook-weight: "1"
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+spec:
+  backoffLimit: 50
+  template:
+    metadata:
+      labels:
+        app: cilium-create-default-policies
+    spec:
+      restartPolicy: OnFailure
+      serviceAccountName: {{ .Values.serviceAccounts.defaultPolicies.name | quote }}
+      priorityClassName: system-cluster-critical
+      {{- with .Values.defaultPolicies.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 6 }}
+      {{- end }}
+      volumes:
+      - name: policies
+        configMap:
+          name: cilium-default-policies
+      initContainers:
+      - name: wait-crds
+        image: "quay.io/giantswarm/docker-kubectl:latest"
+        imagePullPolicy: IfNotPresent
+        command:
+        - sh
+        args:
+        - -c
+        - |
+          for crd in ciliumnetworkpolicies.cilium.io ciliumclusterwidenetworkpolicies.cilium.io
+          do
+            while ! kubectl get crd $crd
+            do
+              echo "Waiting for CRD $crd to exist"
+              sleep 5
+            done
+          done
+      containers:
+      - name: cilium-create-default-policies
+        image: "quay.io/giantswarm/docker-kubectl:latest"
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: policies
+          mountPath: /policies
+        command:
+        - sh
+        args:
+        - -c
+        - |
+          set -o errexit ; set -o xtrace ; set -o nounset
+
+          kubectl apply --server-side=true --field-manager='kubectl-client-side-apply' --force-conflicts -f /policies/ 2>&1
```

Comment:
Will they be removed when the release is deleted?

Reply:
The CRD _should_ be removed when the release is but we'd need to test that as the chart doesn't install the CRDs.

Reply:
according to my tests CRD don't get removed when uninstalling the chart

Reply:
CRDs are not, but policies should be I think

Reply:
any idea how to do so?

Reply:
Maybe helm labels, maye deletion hook

Reply:
ok so, I couldn't find any clean way of doing this.
Also, I don't understand why we should care about deleting those CRs if we don't care about cleaning up the CRDs.
If in the future we'll allow switching between CNI plugins and deleting a CNI app will become a thing, we will need to ensure proper cleanup of CRDs, and CRs will follow automatically.
IMHO it's not worth the effort.

Reply:
I'm not blocking this